Marlin/src/lcd/menu/menu_advanced.cpp

```diff
@@ -416,7 +416,10 @@ void menu_cancelobject();
     #endif
 
     #ifdef XY_FREQUENCY_LIMIT
-      EDIT_ITEM(int8, MSG_XY_FREQUENCY_LIMIT, &planner.xy_freq_limit_hz, 0, 100, planner.refresh_frequency_limit, true);
+      MENU_ITEM_IF(1) {
+        editable.uint8 = planner.xy_freq_limit_hz;
+        EDIT_ITEM(uint8, MSG_XY_FREQUENCY_LIMIT, &editable.uint8, 0, 100, []{ planner.refresh_frequency_limit(); }, true);
+      }
```

Comment:
The value if `editable.uint8` isn't being used in the action handler. Either `xy_freq_limit_hz` must be edited directly, or the value of `editable.uint8` must be used to set `xy_freq_limit_hz` in the action handler.